### PR TITLE
Feature/mf8

### PR DESCRIPTION
**Pull request by whaeck**
Fission yield sections have been added to MF8

PR order:
#123 <-- this one
#120 
#124 
#126
#127

### CODE REVIEW

File: src/ENDFtk/section/8/FissionYieldData/src/generateList.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+static std::vector< double >
+generateList( std::vector< unsigned int >&& identifiers,
+              std::vector< unsigned int >&& states,
+              std::vector< std::array< double, 2 > >&& yields ) {
+
+  unsigned int nfp = identifiers.size();
+  if ( ( states.size() != nfp ) or ( yields.size() != nfp ) ) {
+
+    Log::info( "The number of fission profucts and the fission product data "
```
Comment (nathangibson14) [inline src/ENDFtk/section/8/FissionYieldData/src/generateList.hpp:9]: `Log::error`?

**Review by nathangibson14 — CHANGES_REQUESTED** [review]

File: src/ENDFtk/file/8.hpp [hunk 2]
```diff
@@ -28,18 +28,17 @@ namespace file {
 
     // MT454, MT457 and MT459 are optional
     static constexpr auto optionalSections()
-      RANGES_DECLTYPE_AUTO_RETURN( hana::make_tuple( 457_c ) )
+      RANGES_DECLTYPE_AUTO_RETURN( hana::make_tuple( 454_c, 457_c, 459_c ) )
 
     // the following sections are currently unimplemented
     static constexpr auto unimplementedSections()
-      RANGES_DECLTYPE_AUTO_RETURN( hana::make_tuple( 454_c, 459_c ) )
+      RANGES_DECLTYPE_AUTO_RETURN( hana::make_tuple() )
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/ENDFtk/file/8.hpp:35]: We probably can get rid of it, but I'd like to leave it here for now until we get to implementing the non-enumerated sections for MF8.

File: src/ENDFtk/section/8/454/src/generateFissionYields.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      Log::info( "The number of fission yield values for the fission product "
+                 "with index {} is not equal to the expected value", i );
+      Log::info( "NE value: {}", size );
+      Log::info( "yields[i].size(): {}", yields[i].size() );
+      throw std::exception();
+    }
+  }
+
+  unsigned int i = 0;
+  while ( i++ < size ) {
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/ENDFtk/section/8/454/src/generateFissionYields.hpp:36]: I have no idea why I did it this way. I wrote this months ago ;-)

File: src/ENDFtk/section/8/459/src/generateFissionYields.hpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+static std::vector< FissionYieldData >
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/ENDFtk/section/8/459/src/generateFissionYields.hpp:1]: I removed this.

File: src/ENDFtk/section/8/459/src/NC.hpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/**
+ *  @brief Return the number of lines in this MF7/MT2 section
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/ENDFtk/section/8/459/src/NC.hpp:2]: I removed this.

File: src/ENDFtk/section/8/459.hpp [hunk 6]
```diff
@@ -67,8 +68,9 @@ namespace section {
      */
     auto incidentEnergies() const { return this->E(); }
 
-    #include "ENDFtk/section/8/459/src/NC.hpp"
-    #include "ENDFtk/section/8/459/src/print.hpp"
+    // these deliberately reference MF8/MT454 since the code is common
```
Comment (nathangibson14) [inline src/ENDFtk/section/8/459.hpp:71]: Good comments to add! :)

**Review by nathangibson14 — APPROVED** [review]

**Pull request merged**